### PR TITLE
BUG: ne comparison returns False for NA and other value

### PR DESCRIPTION
**Pull request by phofl**
returning True here makes more sense than False for !=

### CODE REVIEW

File: pandas/core/arrays/string_arrow.py [hunk 1]
```diff
@@ -663,7 +664,10 @@ def _convert_int_dtype(self, result):
 
     def _cmp_method(self, other, op):
         result = super()._cmp_method(other, op)
```
Comment (mroeschke) [inline pandas/core/arrays/string_arrow.py:666]: Shouldn't this be fixed in the `ArrowExtensionArray._cmp_method`?

Comment (phofl) [reply pandas/core/arrays/string_arrow.py:666]: No, the arrow dtypes will return an arrow array with bool dtype that has NA values in those places, which is correct, this fix is to emulate object dtype

**Comment by mroeschke** [comment]
pre-commit.ci autofix

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**